### DISCUSSION TITLE
Setting up a project using liburing with cmake

### DISCUSSION THREAD

**Discussion by fquinsol** [axboe/liburing#576] [Q&A] [answered]
I'm having compilation issues and I wonder what I should be expecting in terms of which version of the library to use.
The specific errors are the bottom of this post
I'm attempting to compile in a WSL instance of Ubuntu 20.04. The kernel version is 5.10.
I have setup a C++ project using cmake toolchain and pull down liburing 2.1 via conan.

For example:
[build] In file included from ../src/main.cpp:6:
[build] /home/wsluser/.conan/data/liburing/2.1/_/_/package/7ba76be63deb70dece6601d6a9a507ed78380172/include/liburing.h: In function ‘void __io_uring_set_target_fixed_file(io_uring_sqe*, unsigned int)’:
[build] /home/wsluser/.conan/data/liburing/2.1/_/_/package/7ba76be63deb70dece6601d6a9a507ed78380172/include/liburing.h:238:7: error: ‘struct io_uring_sqe’ has no member named ‘file_index’
[build]   238 |  sqe->file_index = file_index + 1;
[build]       |       ^~~~~~~~~~

**Comment by axboe** [comment, marked as answer]
You're using the 5.10 kernel io_uring.h header, you should use the one that ships with liburing. I'm guessing your build system changes made it pick up another one rather than the src/include/liburing/io_uring.h it should be using.

**Reply by fquinsol** [reply]
Thanks for your reply.
Now it compiles with some warnings. These below are being redefined: 
[build] In file included from /home/wsluser/.conan/data/liburing/2.1/_/_/package/7ba76be63deb70dece6601d6a9a507ed78380172/include/liburing/io_uring.h:11,
[build]                  from /home/wsluser/.conan/data/liburing/2.1/_/_/package/7ba76be63deb70dece6601d6a9a507ed78380172/include/liburing.h:20,
[build]                  from ../src/main.cpp:5:
[build] /home/wsluser/.conan/data/linux-headers-generic/5.13.9/_/_/package/c0519e2d9702ec12d057bb15adb7a02baaf18107/include/linux/fs.h:285: warning: "RWF_HIPRI" redefined
[build]   285 | #define RWF_HIPRI ((__kernel_rwf_t)0x00000001)
[build]       | 
[build] In file included from /usr/include/x86_64-linux-gnu/sys/uio.h:168,
[build]                  from /home/wsluser/.conan/data/liburing/2.1/_/_/package/7ba76be63deb70dece6601d6a9a507ed78380172/include/liburing.h:10,
[build]                  from ../src/main.cpp:5:
[build] /usr/include/x86_64-linux-gnu/bits/uio-ext.h:45: note: this is the location of the previous definition
[build]    45 | #define RWF_HIPRI 0x00000001 /* High priority request.  */

**Reply by fquinsol** [reply]
I also would want to confirm that although it includes some stuff from kernel 5.13 headers, it would work on kernel 5.10. 

Thanks
